### ISSUE TITLE
Tests and spec seem to be unclear on NumberFormat and types

### ISSUE THREAD

**Issue by tmiddlet2666** [eclipse/microprofile-graphql#209]
In  https://download.eclipse.org/microprofile/microprofile-graphql-1.0/microprofile-graphql.html#_numbers  It states 

> When formatting is added to a number type, the formatted result will be of type String.

The example:
```
@JsonbNumberFormat(value = "¤000",locale = "en-ZA")
private Short formattedShortObject;
```
Should generate a schema of 
```
  "¤000 en-ZA"
  formattedShortObject: String
```

Tests should be added to ensure this is the case in the generated schema.

**Comment by phillip-kruger** [comment]
@andymc12 - can you confirm this ? I know in SmallRye the schema still generated a response of Int for the above example, but that does seems wrong. I would think this apply to all formatted input and output.

**Comment by kaqqao** [comment]
In my current implementation formatted number appear as a custom scalar called `FormattedNumber`.
This is consistent with the 1.0 behavior, but also with the date approach where we have a custom `Date` scalar.

**Comment by tmiddlet2666** [comment]
Ah ok @kaqqao so the resultant schema type should not be validated because it is implementation specific on how formatting is handled? 
I guess the spec is saying that the result should be a string value.

**Comment by andymc12** [comment]
What if we add a TCK test case that just checks the schema shows that the value is not an `Int`?  That would allows implementations to choose a schema type of `String` or custom scalars like `FormattedString`, `StringifiedNumber`, etc.

**Comment by phillip-kruger** [comment]
Fixed in #240
